File: csharp-misinformation/secure9.cs
using OpenAI.Chat;
using Microsoft.AspNetCore.Mvc;

[ApiController]
[Route("[controller]")]
public class AskController : ControllerBase
{
    private const string Disclaimer = "\n\nNote: AI-generated content. Verify independently.";
    private readonly ChatClient _chatClient = new ChatClient("gpt-4o", "API_KEY");

    [HttpPost("ask")]
    public async Task<IActionResult> Ask([FromQuery] string question)
    {
        ChatCompletion completion = await _chatClient.CompleteChatAsync(
            new UserChatMessage(question));
        string answer = completion.Content[0].Text;
        return Content(answer + Disclaimer, "text/plain");
    }
}

File: javascript-misinformation/secure9.js
import { Hono } from "hono";
import OpenAI from "openai";

const app = new Hono();
const openai = new OpenAI();
const DISCLAIMER = "AI-generated content. Verify independently before acting on this information.";

app.post("/ask", async (c) => {
  const { question } = await c.req.json();
  const completion = await openai.chat.completions.create({
    model: "gpt-4o",
    messages: [{ role: "user", content: question }],
  });
  const answer = completion.choices[0].message.content;
  return c.json({ answer, disclaimer: DISCLAIMER });
});

export default app;
